### PR TITLE
Feature/version one

### PR DESCRIPTION
**Pull request by pDiller**
Restructured the complete application.

### CODE REVIEW

File: src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	private static final String ALERT_CLASS = "alert";
+	private static final String ALERT_DANGER_CLASS = "alert-danger";
+	private boolean hasErrorMessages;
+
+	@Override
+	public void onConfigure(Component component) {
+		super.onConfigure(component);
+
+		if (!(component instanceof FeedbackPanel)) {
+			throw new IllegalArgumentException("Dieses Behavior darf ausschliesslich auf FeedbackPanels angewendet werden.");
```
Comment (danielvdwal) [inline src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java]: German exception messages?

File: src/main/java/io/reflectoring/jiraalerts/common/FormComponentErrorClassModifier.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	private static final String INVALID_CLASS = "is-invalid";
+	private static final String FORM_CONTROL_CLASS = "form-control";
+	private boolean valid;
+
+	@Override
+	public void onConfigure(Component component) {
+		super.onConfigure(component);
+
+		if (!(component instanceof FormComponent)) {
+			throw new IllegalArgumentException("Dieses Behavior darf ausschliesslich auf FeedbackPanels angewendet werden.");
```
Comment (danielvdwal) [inline src/main/java/io/reflectoring/jiraalerts/common/FormComponentErrorClassModifier.java]: German exception messages?

File: src/main/java/io/reflectoring/jiraalerts/login/HashService.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+			MessageDigest md = MessageDigest.getInstance(SHA_512);
+			md.update(salt.getBytes(STRING_ENCODING));
+			byte[] bytes = md.digest(password.getBytes(STRING_ENCODING));
+			StringBuilder sb = new StringBuilder();
+			for (int i = 0; i < bytes.length; i++) {
+				sb.append(Integer.toString((bytes[i] & 0xff) + 0x100, 16).substring(1));
+			}
+			return sb.toString();
+		} catch (NoSuchAlgorithmException | UnsupportedEncodingException exception) {
+			throw new IllegalStateException("Beim hashen des Passwortes ist ein Fehler aufgetreten: ", exception);
```
Comment (danielvdwal) [inline src/main/java/io/reflectoring/jiraalerts/login/HashService.java]: German exception messages?

File: src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java (hunk 1 above)
Comment (pDiller) [reply src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java]: Sorry, i´ve been in coding mode :-)

File: src/main/java/io/reflectoring/jiraalerts/common/FormComponentErrorClassModifier.java (hunk 2 above)
Comment (pDiller) [reply src/main/java/io/reflectoring/jiraalerts/common/FormComponentErrorClassModifier.java]: Sorry, i´ve been in coding mode :-)

File: src/main/java/io/reflectoring/jiraalerts/login/HashService.java (hunk 3 above)
Comment (pDiller) [reply src/main/java/io/reflectoring/jiraalerts/login/HashService.java]: Sorry, i´ve been in coding mode :-)

File: src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	private static final String ALERT_DANGER_CLASS = "alert-danger";
+	private boolean hasErrorMessages;
+
+	@Override
+	public void onConfigure(Component component) {
+		super.onConfigure(component);
+
+		if (!(component instanceof FeedbackPanel)) {
+			throw new IllegalArgumentException(
+			        format("This behavior should only be added to FeedbackPanels.Can´t add behavior to class '%s'", component.getClass()));
```
Comment (danielvdwal) [inline src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java]: There is a space missing

File: gradle/sonarqube.gradle [hunk 5]
```diff
@@ -0,0 +1,25 @@
+apply plugin: 'org.sonarqube'
+apply plugin: 'jacoco'
+
+sonarqube {
+    properties {
+        property "sonar.projectName", "JiraAlerts"
+        property "sonar.projectKey", "io.reflectoring.jiraalerts:JiraAlerts-Sonar"
+        property "sonar.host.url", "https://sonarcloud.io"
+        property "sonar.organization", "pdiller-github"
+        property "sonar.login", "6dedb17d06baa59bd81b5e936e192f11526a47b8"
```
Comment (danielvdwal) [inline gradle/sonarqube.gradle]: there must be a way to hide this information. Isn't there? Like entering this inside the ide?

File: src/main/resources/io/reflectoring/jiraalerts/dashboard/ApplicationStatusDashboardCardPanel.properties [hunk 6]
```diff
@@ -0,0 +1,2 @@
+card.header.text=Applikations Status
```
Comment (danielvdwal) [inline src/main/resources/io/reflectoring/jiraalerts/dashboard/ApplicationStatusDashboardCardPanel.properties]: Applikationsstatus?

File: src/main/resources/io/reflectoring/jiraalerts/dashboard/UserManagementDashboardCardPanel.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+card.header.text=Benutzerverwaltung
+card.body.text=Hier kannst du weiter Nutzer anlegen und deren Rechte bearbeiten.
```
Comment (danielvdwal) [inline src/main/resources/io/reflectoring/jiraalerts/dashboard/UserManagementDashboardCardPanel.properties]: *weitere
*Rechte verwalten?

Comment (pDiller) [reply src/main/resources/io/reflectoring/jiraalerts/dashboard/UserManagementDashboardCardPanel.properties]: good point. fixed it  whit next commit.

File: src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java (hunk 4 above)
Comment (pDiller) [reply src/main/java/io/reflectoring/jiraalerts/common/FeedbackPanelErrorClassModifier.java]: good point. fixed it  whit next commit.

File: gradle/sonarqube.gradle (hunk 5 above)
Comment (pDiller) [reply gradle/sonarqube.gradle]: I think it´s ok for now. If we move the responsibility of this app to reflectoring organisation, we have to rethink the build strategy.

**Pull request merged**